src/mjolnir/pbfgraphparser.cc

```diff
@@ -5173,20 +5195,27 @@ void PBFGraphParser::ParseNodes(const boost::property_tree::ptree& pt,
 
     bool create = true;
     for (auto& file : input_files) {
-      callback.current_way_node_index_ = callback.last_node_ = callback.last_way_ =
-          callback.last_relation_ = 0;
+      parser.current_way_node_index_ = parser.last_node_ = parser.last_way_ = parser.last_relation_ =
+          0;
       // we send a null way_nodes file so that only the bike share stations are parsed
-      callback.reset(nullptr, nullptr, nullptr, nullptr, nullptr,
-                     new sequence<OSMNode>(bss_nodes_file, create), nullptr);
-      OSMPBF::parse(file, static_cast<OSMPBF::Interest>(OSMPBF::Interest::NODES), callback);
+      parser.reset(nullptr, nullptr, nullptr, nullptr, nullptr,
+                   new sequence<OSMNode>(bss_nodes_file, create), nullptr);
       create = false;
+
+      osmium::io::Reader reader(file, osmium::osm_entity_bits::node);
+      while (const osmium::memory::Buffer buffer = reader.read()) {
```

Comment:
i wonder how this works actually, i havent read the api but does the reader keep an internal buffer and we copy it out here or are things move assigned or what is going on here? i assume its chunked like it never reads more than N entries at a time so we dont have any risk of OOM but do wonder exactly how memory moves around here.

Reply:
thanks yeah i didnt have time to go look myself, the brief synopsis is much appreciated!